analyzer/src/main/kotlin/managers/utils/NpmDependencyHandler.kt

```diff
@@ -47,7 +47,26 @@ data class NpmModuleInfo(
 
     /** A set with information about the modules this module depends on. */
     val dependencies: Set<NpmModuleInfo>
-)
+) {
+    /**
+     * [workingDir] and [packageFile] are not relevant when adding this [NpmModuleInfo] to the dependency graph.
+     * However, if these values differ the same dependencies are added as duplicates to the set which is used to create
+     * the dependency graph. Therefore, remove them from the equals check.
+     */
+    override fun equals(other: Any?): Boolean {
```

Comment:
One idea as an alternative to overriding `equals()` could be to not use a `Set` in `Npm.getModuleInfo()` when determining the dependencies, but a `Map` using a combined key with only the relevant properties (the identifier and the dependencies). I am not sure, however, whether this is really a significant improvement over the current approach.